### PR TITLE
Add option to ImageOptimizer to skip compressing files in unsupported formats instead of throwing an exception

### LINKED ISSUE #269: Add option to ImageOptimizer to disable throwing exception for unsupported formats.
An option to disable throwing exceptions when a format is unsupported should be added. With this option enabled unsupported files will just be skipped instead. Feel free to suggest a name for the property in this issue.

### PR DESCRIPTION
**Pull request by zachesposito**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/dlemstra/Magick.NET/pulls) open
- [x] I have verified that I am following the existing coding patterns and practices as demonstrated in the repository.

### Description
Add a `IgnoreUnsupportedFormats` option to ImageOptimizer, which when set to true causes the compression methods to return false (indicating compression failed) for unsupported files, instead of throwing an exception.

Fixes #269

### CODE REVIEW

File: Source/Magick.NET/Shared/Optimizers/ImageOptimizer.cs [hunk 1]
```diff
@@ -35,6 +35,15 @@ public bool OptimalCompression
             set;
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to skip unsupported files instead of throwing an exception.
+        /// </summary>
+        public bool IgnoreUnsupportedFormats
```
Comment (dlemstra) [inline Source/Magick.NET/Shared/Optimizers/ImageOptimizer.cs]: The properties and methods are sorted alphabetically, could you move this?

File: Source/Magick.NET/Shared/Optimizers/ImageOptimizer.cs [hunk 2]
```diff
@@ -252,7 +261,10 @@ private IImageOptimizer GetOptimizer(FileInfo file)
                     return optimizer;
             }
 
-            throw new MagickCorruptImageErrorException("Invalid format, supported formats are: " + SupportedFormats);
+            if (IgnoreUnsupportedFormats)
+                return null;
+            else
```
Comment (dlemstra) [inline Source/Magick.NET/Shared/Optimizers/ImageOptimizer.cs]: I think I prefer this without the else, we don't need it here.

File: Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs [hunk 3]
```diff
@@ -448,6 +448,23 @@ public void Compress_CannotCompressPngFile_FileIsNotSmaller()
             });
         }
 
+        [TestMethod]
+        public void Compress_IgnoreUnsupportedFile_DoesNotThrowException()
```
Comment (dlemstra) [inline Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs]: Thanks for following the pattern that is used in this file. Would love to change these to the new unit test structure but that should probably be a separate PR. Would you be willing to change that in a new PR?

File: Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs [hunk 4]
```diff
@@ -448,6 +448,23 @@ public void Compress_CannotCompressPngFile_FileIsNotSmaller()
...
+            Assert.IsFalse(optimizer.Compress(Files.InvitationTif));
+        }
+
+        [TestMethod]
+        public void Compress_IgnoreUnsupportedStream_DoesNotThrowException()
+        {
+            var optimizer = new ImageOptimizer { IgnoreUnsupportedFormats = true };
+            using (FileStream fileStream = OpenFile(Files.InvitationTif))
+            {
+                Assert.IsFalse(optimizer.Compress(fileStream));
```
Comment (dlemstra) [inline Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs]: I think I would like to have the result of the compress method inside a variable first and assert that. But it seems I did not do that earlier. Probably because I had no idea what the name of the variable would be. Any ideas?

File: Source/Magick.NET/Shared/Optimizers/ImageOptimizer.cs (hunk 1 above)
Comment (zachesposito) [reply Source/Magick.NET/Shared/Optimizers/ImageOptimizer.cs]: Yes the properties are now in correct order

File: Source/Magick.NET/Shared/Optimizers/ImageOptimizer.cs (hunk 2 above)
Comment (zachesposito) [reply Source/Magick.NET/Shared/Optimizers/ImageOptimizer.cs]: Sure thing, elses now removed

File: Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs (hunk 3 above)
Comment (zachesposito) [reply Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs]: Is the new unit test structure like [this test class](https://github.com/dlemstra/Magick.NET/blob/7e103d6bfdbcef967d5d9e97ecbb31dd3b5037a9/Tests/Magick.NET.Tests/Windows/Defines/Pdf/PdfReadDefinesTests/ThePasswordProperty.cs#L27), where the class name is "The[ThingBeingTested]" and the method names are "Should[ExpectedBehavior]"? If so yes I can update these tests in a new PR.

File: Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs (hunk 4 above)
Comment (zachesposito) [reply Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs]: We could call it `compressionSuccess`:

```
[TestMethod]
public void Compress_IgnoreUnsupportedStream_DoesNotThrowException()
{
    var optimizer = new ImageOptimizer { IgnoreUnsupportedFormats = true };
    using (FileStream fileStream = OpenFile(Files.InvitationTif))
    {
        var compressionSuccess = optimizer.Compress(fileStream);
        Assert.IsFalse(compressionSuccess);
    }
}
```

Comment (dlemstra) [reply Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs]: Sound like a good name :)

Comment (dlemstra) [reply Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs]: Could you do this in the PR where you refactor the tests?

Comment (zachesposito) [reply Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs]: Sure thing I will include this change in the next PR

File: Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs (hunk 3 above)
Comment (zachesposito) [reply Tests/Magick.NET.Tests/Shared/Optimizers/ImageOptimizerTests.cs]: OK I will follow that pattern in the next PR
